epictrack-web/src/components/project/ProjectForm.tsx

```diff
@@ -268,6 +285,54 @@ export default function ProjectForm({ ...props }) {
                 {...register("proponent_id")}
               ></ControlledSelectV2>
             </Grid>
+            <Grid item xs={12}>
+              <When condition={Boolean(specialField)}>
+                <SpecialFieldGrid
+                  entity={SpecialFieldEntityEnum.PROJECT}
+                  entity_id={(ctx.item as Project)?.id}
+                  fieldName={specialField}
+                  fieldLabel={
+                    specialField === "proponent_id" ? "Proponent Name" : "Name"
```

Comment:
In this design, what happens if we have more than 2 special fields. Can we refactor this to be more scalable?